src/main/cpp/src/hive_hash.cu

```diff
@@ -389,18 +389,17 @@ class hive_device_row_hasher {
     __device__ hive_hash_value_t hash_nested(cudf::size_type col_index,
                                              cudf::size_type row_index) const noexcept
     {
-      auto curr_col_idx = _parent._column_map[col_index];
-      auto next_col_idx = curr_col_idx;
-      auto curr_row_idx = row_index;
+      auto const flattened_idx = _parent._column_map[col_index];
+      auto next_col_idx        = flattened_idx + 1;
```

Comment:
Why there is a  `next_col_idx`  not in the `while` block, all the information for computing should be poped in the  `while` block. This will probably cause errors.

Reply:
Seems `col_stack_frame` does not contain all the information we need when using prefix iteration.
```cpp
    struct col_stack_frame {
      cudf::size_type _col_idx;     // the column index in the flattened array
      cudf::size_type _row_idx;     // the index of the row in the column
      int _idx_to_process;          // the index of child or element to process next
      hive_hash_value_t _cur_hash;
    };
```
If we use level by level iteration, then we can easily get the next direct child from `_col_idx` and `_idx_to_process`:
```
col_infos[_col_idx + _idx_to_process]
```
but for prefix iteration, it's hard to get the next direct child from a `col_stack_frame`.

Reply:
Sorry for the previous comment, the current solution works, and it uses less memories.